service/forward_service.cc

```diff
@@ -542,11 +542,7 @@ future<query::forward_result> forward_service::dispatch(query::forward_request r
                             flogger.debug("received forward_result={} from {}", partial_result_printer, addr);
                             
                             return aggrs.with_thread_if_needed([&result_, &aggrs, partial_result = std::move(partial_result)] () mutable {
```

Comment:
As a follow up, since `req` is known upon entry, I think we could run `with_thread_if_needed` outside parallel_for_each saving the need to launch a thread for each merge.

Reply:
But we need the thread in the body of parallel_for_each, that's where the code that requires threads is used. Calling parallel_for_each from inside a thread doesn't make the parallel_for_each body execute in threaded context.